tests/diagnostics/switch-mutable-cases.slang

```diff
@@ -0,0 +1,16 @@
+//TEST:SIMPLE(filecheck=CHECK):
+
+// Tests to evaluate the behavior of code blocks within a switch statement. A switch statement with duplicate cases with same values is not allowed and should throw an error
+
+void test()
+{
+    int a = 1;
+    int b = 2;
+    switch (0) {
+        // CHECK: ([[# @LINE+1]]): error 39999: expression {{.*}}
+        case a: break;
```

Comment:
case a shouldn't be a valid input code.

Should be case 0.

In the future we should add a diagnostic if Folding to Val failed.

Reply:
this was a contrived test case that is to catch the case when const-folding to int val would fail. This is automatically caught by `slang-check-expr.cpp:CheckIntegerConstantExpression`, which is why I didn't want to add another diagnostic in `visitSwitchStmt`.

If I have const int for cases, then it wouldn't fail at `tryFoldIntegerConstantExpression` within that function.

do you want me to remove this test case, or can we leave it as a negative test?

Reply:
OK, let's leave it here.